pulumi/infra/kafka.py

```diff
@@ -1,13 +1,137 @@
-from typing import Optional
+from __future__ import annotations
 
-import pulumi_kafka as kafka
+import dataclasses
+import os
+from typing import Any, Iterable, Mapping, Optional, Sequence, Tuple, cast
+
+from infra.config import LOCAL_GRAPL
+from pulumi.stack_reference import StackReference
 
 import pulumi
 
+# this list of service names must match those in the
+# confluent-cloud-infrastructure project:
+# TODO: link
+KAFKA_SERVICES = [
+    "pipeline-ingress",
+    "graph-generator",
+    "node-identifier",
+    "graph-merger",
+    "analyzer-executor",
+    "engagement-creator",
+]
+
+
+@dataclasses.dataclass
+class CredentialOutput:
+    service_account_id: str
+    api_key: str
+    api_secret: str
+
+    @staticmethod
+    def from_json(json_: Mapping[str, str]) -> CredentialOutput:
+        return CredentialOutput(
+            service_account_id=json_["service_account_id"],
+            api_key=json_["api_key"],
+            api_secret=json_["api_secret"],
+        )
+
+
+@dataclasses.dataclass
+class TopicOutput:
+    partitions: int
+    config: Mapping[str, Any]
+
+    @staticmethod
+    def from_json(json_: Mapping[str, Any]) -> TopicOutput:
+        return TopicOutput(
+            partitions=json_["partitions"],
+            config=json_["config"],
+        )
+
+
+@dataclasses.dataclass
+class ServiceOutput:
+    ingress_topics: Sequence[str]
+    egress_topics: Sequence[str]
+    service_account: CredentialOutput
+
+    @staticmethod
+    def from_json(json_: Mapping[str, Any]) -> ServiceOutput:
+        return ServiceOutput(
+            ingress_topics=json_["ingress_topics"],
+            egress_topics=json_["egress_topics"],
+            service_account=CredentialOutput.from_json(json_["service_account"]),
+        )
+
+
+@dataclasses.dataclass
+class EnvironmentOutput:
+    environment_id: str
+    bootstrap_servers: str
+    environment_credentials: CredentialOutput
+    services: Mapping[str, ServiceOutput]
+    topics: Mapping[str, TopicOutput]
+
+    def get_service_credentials(self, service_name: str) -> CredentialOutput:
+        if service_name in self.services:
+            return self.services[service_name].service_account
+        else:
+            raise KeyError(f"{service_name} does not exist")
+
+    @staticmethod
+    def from_json(json_: Mapping[str, Any]) -> EnvironmentOutput:
+        return EnvironmentOutput(
+            environment_id=json_["environment_id"],
+            bootstrap_servers=json_["bootstrap_servers"],
+            environment_credentials=CredentialOutput.from_json(
+                json_["environment_credentials"]
+            ),
+            services={
+                k: ServiceOutput.from_json(v) for k, v in json_["services"].items()
+            },
+            topics={k: TopicOutput.from_json(v) for k, v in json_["topics"].items()},
+        )
+
+
+@dataclasses.dataclass
+class ConfluentOutput:
```

Comment:
Curious why all of these dataclasses are named like `FooOutput`; it seems like just `Foo` would suffice. Am I missing something?